src/Microsoft.Health.Fhir.Core/Features/Operations/ConvertData/ConvertDataEngine.cs

```diff
@@ -95,7 +95,15 @@ private string GetConvertDataResult(ConvertDataRequest convertRequest, ITemplate
                     throw new ConvertDataTimeoutException(Core.Resources.ConvertDataOperationTimeout, convertException.InnerException);
                 }
 
-                _logger.LogInformation(convertException, "Convert data failed.");
+                if (convertException is PostprocessException)
+                {
+                    _logger.LogInformation($"An exception of type {convertException.GetType()} occurred during postprocessing. {convertException.StackTrace}", "Convert data failed.");
+                }
+                else
+                {
+                    _logger.LogInformation(convertException, "Convert data failed.");
+                }
+
                 throw new ConvertDataFailedException(string.Format(Core.Resources.ConvertDataFailed, convertException.Message), convertException);
```

Comment:
You might want to look in the OperationOutcomeExceptionFilterAttribute to make sure the exception message isn't getting logged there.

Reply:
Confirmed with Will this is a 400 response code so it wouldn't get logged there.

Reply:
For reference, this is the line of code that handles the ConvertDataFailedException as an HTTP 400: https://github.com/microsoft/fhir-server/blob/00facae70124c40d5f05fcd26fd504f98cbaced5/src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs#L159

Which means that the code that logs exceptions >= HTTP 500 would not be executed here:
https://github.com/microsoft/fhir-server/blob/00facae70124c40d5f05fcd26fd504f98cbaced5/src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs#L269C25-L269C25